### ISSUE TITLE
Unable to debug an OpenAI baselines pipeline task

### ISSUE THREAD

**Issue by davidhopper2003** [Microsoft/ptvsd#1058]
I used VSCode to debug the PPO2 algorithm in the project [OpenAI baselines](https://github.com/openai/baselines). Lines 57-58  of the file `baselines/common/vec_env/subproc_vec_env.py` uses the pipeline to receive data. When debugging to line 58:
```python
observation_space, action_space = self.remotes[0].recv()
```
the program has no response and the debugging process cannot continue (As shown below). 

![ppo2_debug_1](https://user-images.githubusercontent.com/31435476/49198209-04d95f80-f3cd-11e8-9d36-3f7028c23c09.png)
![ppo2_debug_2](https://user-images.githubusercontent.com/31435476/49198210-04d95f80-f3cd-11e8-84a4-cb2250ec863c.png)
![ppo2_debug_3](https://user-images.githubusercontent.com/31435476/49198211-0571f600-f3cd-11e8-8b33-90b7e52941f4.png)

My configuration file `.vscode/launch.json` is shown as follows:

```
{

    // Use IntelliSense to learn about possible attributes.

    // Hover to view descriptions of existing attributes.

    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387

    "version": "0.2.0",

    "configurations": [

        {

            "name": "Python: Module",

            "type": "python",

            "request": "launch",

            "module": "baselines.run",

            "console": "integratedTerminal",

            "args": [

                "--alg",

                "ppo2",

                "--env",

                "PongNoFrameskip-v4",

                "--num_timesteps",

                "2e7",

                "--save_path",

                "~/models/pong_20M_ppo2"

            ],

        },

        {

            "name": "Python: Current File (Integrated Terminal)",

            "type": "python",

            "request": "launch",

            "program": "${workspaceFolder}/baselines/run.py",

            "console": "integratedTerminal", 

            "args": [

                "--alg",

                "ppo2",

                "--env",

                "PongNoFrameskip-v4",

                "--num_timesteps",

                "2e7",

                "--save_path",

                "~/models/pong_20M_ppo2"

            ],

        }

    ]

}
```

When I used the command line 

`python -m baselines.run --alg=ppo2 --env=PongNoFrameskip-v4 --num_timesteps=2e7 --save_path=~/models/pong_20M_ppo2` 

to run the program in a terminal, there is no problem. 

Any helpful suggestions will be highly appreciated. Thank you.

**Comment by karthiknadig** [comment]
@davidhopper2003 Can you add `"subProcess": true` to the launch arguments, and see if that helps?

**Comment by davidhopper2003** [comment]
@karthiknadig Thanks for your advice. I've tried it but it doesn't work.
![launch_modify](https://user-images.githubusercontent.com/31435476/49265338-a96ea680-f48c-11e8-9c87-87f2554c38bd.png)
![ppo2_debug_4](https://user-images.githubusercontent.com/31435476/49265344-af648780-f48c-11e8-8cca-83a51b80c794.png)
![ppo2_debug_5](https://user-images.githubusercontent.com/31435476/49265347-affd1e00-f48c-11e8-9c40-f09d1ccf4457.png)

**Comment by karthiknadig** [comment]
I think this might be due to `multiprocessing` using `fork`. That should be addressed by this issue #943. 
You could try set `multiprocessing.set_start_method('spawn')` before using it.

**Comment by davidhopper2003** [comment]
@karthiknadig 
This method does work. Great! You're awesome. Thank you very much. Here is my modifications in the file `baselines/common/vec_env/subproc_vec_env.py`: 
#### Modification 1: Line 1-3
```
import numpy as np
from multiprocessing import Process, Pipe
from . import VecEnv, CloudpickleWrapper
```
to 
```
import numpy as np
import multiprocessing
from multiprocessing import Process, Pipe
from . import VecEnv, CloudpickleWrapper
``` 
#### Modification 2: Line 48-50
```
self.remotes, self.work_remotes = zip(*[Pipe() for _ in range(nenvs)])
self.ps = [Process(target=worker, args=(work_remote, remote, CloudpickleWrapper(env_fn)))
                for (work_remote, remote, env_fn) in zip(self.work_remotes, self.remotes, env_fns)]
```
to 
```
self.remotes, self.work_remotes = zip(*[Pipe() for _ in range(nenvs)])

# Set the multi-process startup method to 'spawn` instead of `fork` to support VSCode debugging.
multiprocessing.set_start_method('spawn')

self.ps = [Process(target=worker, args=(work_remote, remote, CloudpickleWrapper(env_fn)))
                for (work_remote, remote, env_fn) in zip(self.work_remotes, self.remotes, env_fns)]
``` 
![ppo2_debug_6](https://user-images.githubusercontent.com/31435476/49272882-0d559700-f4ae-11e8-8a5b-debae378eda0.png)
